internal/challenge/fixme/basic/basic_test.go

```diff
@@ -39,8 +39,15 @@ func TestClosedChannelWithoutOkCheck(t *testing.T) {
 
 	for {
 		select {
-		case val := <-ch:
+		case val, ok := <-ch:
 			slog.Info("received", "value", val)
+			if !ok {
+				ch = nil
```

Comment:
You shouldn't set the channel to nil, it might cause panics elsewhere in the code.